dask/dataframe/core.py

```diff
@@ -3852,8 +3915,9 @@ class DataFrame(_Frame):
     _token_prefix = "dataframe-"
     _accessors = set()
 
-    def __init__(self, dsk, name, meta, divisions):
-        super().__init__(dsk, name, meta, divisions)
+    def __init__(self, expr):
+        super().__init__(expr)
+        name = self._name
```

Comment:
This API change does make sense.  However, I fear that there is a lot of down-stream code with explicit `DataFrame()` usage.  Perhaps we can use something like `def __init__(self, *args):`, and temporarily preserve "legacy"-API handling (with a clear deprecation warning)?

Reply:
That's what we've already done with new_dd_object, so it wouldn't be hard to do it here too. I don't want to add any deprecations yet because we definitely haven't settled on this particular API design yet.

Reply:
That makes sense.  Thanks for clarifying!